### PR TITLE
Add event logging

### PR DESCRIPTION
**Pull request by ShishirH**
Signed-off-by: Shishir Halaharvi <shishir.neo95@gmail.com>

### CODE REVIEW

File: src/main/java/com/autotune/utils/KubeEventLogger.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import io.fabric8.kubernetes.api.model.ObjectReference;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Clock;
+import java.time.Instant;
+
+public class KubeEventLogger implements EventLogger {
```
Comment (dinogun) [inline src/main/java/com/autotune/utils/KubeEventLogger.java:30]: Please add a javadoc here

**Review by dinogun — COMMENTED** [review]
This is looking good! A few minor changes

File: src/main/java/com/autotune/utils/EventLogger.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2021 Red Hat, IBM Corporation and others.
```
[the remark this answers is not in the document]
Comment (ShishirH) [reply src/main/java/com/autotune/utils/EventLogger.java]: Updated the year

File: autotune.iml [hunk 3]
```diff
@@ -1,72 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (ShishirH) [reply autotune.iml]: Restored the file

File: src/main/java/com/autotune/utils/KubeEventLogger.java (hunk 1 above)
Comment (ShishirH) [reply src/main/java/com/autotune/utils/KubeEventLogger.java:34]: Added a javadoc detailing what the class does

File: tests/scripts/common/common_functions.sh [hunk 4]
```diff
@@ -847,8 +847,9 @@ function create_expected_liststacktunables_json() {
 
 		printf '\n    "hpo_algo_impl":  "'${hpo_algo_impl}'",' >> ${file_name}
 
-		printf '\n    "deployment_name": "'${deployment_names[index]}'",' >> ${file_name}
-		printf '\n    "namespace":  '$(cat ${autotune_json} | jq '.metadata.namespace')',' >> ${file_name}
+		# Uncomment these when the json output is updated
```
Comment (chandrams) [inline tests/scripts/common/common_functions.sh:850]: Why is this file included? @ShishirH - Can you please check and re-base it so that it includes only the relevant files.

Comment (ShishirH) [reply tests/scripts/common/common_functions.sh]: Sorry for that! Removed the file and rebased it

**Review by dinogun — APPROVED** [review]
LGTM

**Review by chandrams — APPROVED** [review]
Thanks @ShishirH, looks good now
